src/backend/distributed/planner/multi_explain.c

```diff
@@ -203,7 +203,10 @@ CitusExplainScan(CustomScanState *node, List *ancestors, struct ExplainState *es
 
 	if (distributedPlan->subPlanList != NIL)
 	{
+		/* ExplainSubPlans calls ExplainOnePlan which uses the ActiveSnapshot.*/
```

Comment:
I wonder whether we should push snaphost at the start of CitusExplainScan and pop at the end of the function, such that all the parts of the explain has the same snapshot?

Reply:
Sure. I see no harm to enclose all the Explain functions called here.